iceoryx_posh/BUILD.bazel

```diff
@@ -25,18 +26,57 @@ configure_file(
     name = "iceoryx_posh_deployment_hpp",
     src = "cmake/iceoryx_posh_deployment.hpp.in",
     out = "include/iceoryx_posh/iceoryx_posh_deployment.hpp",
-    config = {
+    config = select({
         # FIXME: for values see "iceoryx_posh/cmake/IceoryxPoshDeployment.cmake" ... for now some nice defaults
-        "IOX_COMMUNICATION_POLICY": "ManyToManyPolicy",
-        "IOX_INTERNAL_MAX_NUMBER_OF_NOTIFIERS": "256",
-        "IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY": "8",
-        "IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY": "256",
-        "IOX_MAX_INTERFACE_NUMBER": "4",
-        "IOX_MAX_PUBLISHERS": "512",
-        "IOX_MAX_PUBLISHER_HISTORY": "16",
-        "IOX_MAX_SUBSCRIBERS": "1024",
-        "IOX_MAX_SUBSCRIBERS_PER_PUBLISHER": "256",
-    },
+        "@platforms//os:macos": {
+            "IOX_COMMUNICATION_POLICY": "ManyToManyPolicy",
+            "IOX_INTERNAL_MAX_NUMBER_OF_NOTIFIERS": "256",
+            "IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY": "8",
+            "IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY": "256",
+            "IOX_MAX_CLIENTS_PER_SERVER": "256",
+            "IOX_MAX_ID_STRING_LENGTH": "100",
+            "IOX_MAX_INTERFACE_NUMBER": "4",
+            "IOX_MAX_NODE_NAME_LENGTH": "100",
```

Comment:
OK, we add more fine grained control over some parameters (such as `IOX_MAX_NODE_NAME_LENGTH`).

Reply:
Shouldn't this be 98 here?